src/Foundatio/Utility/ScheduledTimer.cs

```diff
@@ -61,7 +63,8 @@ public class ScheduledTimer : IDisposable {
 
                 _logger.Trace(() => $"Scheduling next: delay={delay}");
 
-                _timer.Change(delay, Timeout.Infinite);
+                DisposeTimer();
+                _timer = SystemClock.Instance.Schedule(TimeSpan.FromMilliseconds(delay), async () => await RunCallbackAsync());
```

Comment:
What about a race condition between when schedule next is called a bunch? I think that's one of the reasons we were calling change on the timer.